app/data_model/answer_store.py

```diff
@@ -245,3 +245,47 @@ def natural_order(key):
     :return:
     """
     return [number_else_string(c) for c in re.split(r'(\d+)', key)]
+
+
+def upgrade_0_to_1_update_date_formats(answer_store, schema):
+    """ Updates the date format """
+    for answer in answer_store.answers:
+        answer_schema = schema.get_answer(answer['answer_id'])
+
+        if answer_schema:
+            if answer_schema['type'] == 'Date':
+                answer['value'] = datetime.strptime(answer['value'], '%d/%m/%Y').strftime('%Y-%m-%d')
+                continue
+
+            if answer_schema['type'] == 'MonthYearDate':
+                answer['value'] = datetime.strptime(answer['value'], '%m/%Y').strftime('%Y-%m')
+                continue
+
+
+def upgrade_1_to_2_add_group_instance_id(answer_store, schema):
+    """ Answers should have a `group_instance_id` ready for more complex group repeat rules. """
+    from app.questionnaire.location import Location
+    from app.helpers.schema_helpers import get_group_instance_id
+
+    for answer in answer_store.answers:
+        # Happily, parent_id's are patched on to the schema for each nested level, so we can
+        # retrieve the block_id and group_id for the answer we're processing here.
+        answer_schema = schema.get_answer(answer['answer_id'])
+        question = schema.get_question(answer_schema['parent_id'])
+        block_id = question['parent_id']
+        block = schema.get_block(question['parent_id'])
+        group_id = block['parent_id']
+
+        location = Location(
+            group_id=group_id,
+            group_instance=answer['group_instance'],
+            block_id=block_id,
+        )
+        # `get_group_instance_id` handles providing a consistent group_instance_id
+        answer['group_instance_id'] = get_group_instance_id(schema, answer_store, location)
+
+
+UPGRADE_TRANSFORMS = (
```

Comment:
My thought was to try and do the most simple thing for now. I think that longer term a means to upgrade other parts of the questionnaire store is required too - potentially the upgrading should happen in the `QuestionnaireStore`, since that's where the version number is stored, and it has access to the other bits of data which **could** need updating at some point (e.g. the metadata would need upgrading if you change a type or add a new claim).

I didn't want to drag us down that rabbit hole as part of this though, I felt this was the minimum required to make it obvious what the upgrades are and will make it easy to patch in a couple more updates if needed (without a refactor), but it might be worth looking at refactoring the store upgrader a bit sooner